### PR TITLE
Merge task_force_hotfix to new_dawn_uat (display precision fix)

### PR DESCRIPTION
**Pull request by teosarca**
## Summary
Merge `task_force_hotfix` → `new_dawn_uat` to port the mobile UI display precision fix.

Original PR: https://github.com/metasfresh/metasfresh/pull/23330

## Conflict resolution
- `JsonCreateProductRequest.java`: kept both `pickingInstruction` (new_dawn_uat) and `issuingTolerancePerc` (hotfix)
- `CreateProductCommand.java`: kept both handling blocks
- `manufacturing_small_qty_tolerance.spec.js`: took hotfix version (latest, includes display assertions)

### CODE REVIEW

File: misc/services/mobile-webui/mobile-webui-frontend/src/utils/qtys.js [hunk 1]
```diff
@@ -51,14 +60,17 @@ const getDefaultDisplayPrecision = (uom, defaultPrecision = 4) => {
 export const formatQtyToHumanReadable = ({ qty, uom, precision = null }) => {
   let qtyEffective = qty ?? 0;
   let uomEffective = uom;
+  let decimalShift = 0;
 
   while (qtyEffective !== 0 && Math.abs(qtyEffective) < 1) {
     if (uomEffective === 'kg') {
```
[the remark this answers is not in the document]
Comment (teosarca) [reply misc/services/mobile-webui/mobile-webui-frontend/src/utils/qtys.js]: Fixed in https://github.com/metasfresh/metasfresh/pull/23336 — normalized qty to Number upfront. Added regression test for string '0' and other string edge cases (57 Jest tests total).

File: misc/services/mobile-webui/mobile-webui-frontend/src/__tests__/utils/qtys.test.js [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+      // Normal kg quantities (no conversion, qty >= 1)
+      [{ qty: 100, uom: 'kg' }, '100', 'kg'], // 0 decimals, no shift
+      [{ qty: 1.5, uom: 'kg' }, '1.5', 'kg'], // 1 decimal, no shift
+      [{ qty: 2.25, uom: 'kg' }, '2.25', 'kg'], // 2 decimals, no shift
+
+      // Non-kg UOM (no conversion possible)
+      [{ qty: 0.5, uom: 'PCE' }, '0.5', 'PCE'], // 1 decimal, no shift, default precision max(1,4)=4 → but only 1 significant decimal
+
+      // Explicit precision overrides derived precision
+      [{ qty: 0.019125, uom: 'kg', precision: 999 }, '19.125', 'g'], // explicit 999 → shows all
```
[the remark this answers is not in the document]
Comment (teosarca) [reply misc/services/mobile-webui/mobile-webui-frontend/src/__tests__/utils/qtys.test.js]: Fixed in https://github.com/metasfresh/metasfresh/pull/23336 — updated comment to reflect the cap.
